### DISCUSSION TITLE
Entity.Get<>()  Problem

### DISCUSSION THREAD

**Discussion by demian-moon** [genaray/Arch#235] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/b9e6e436-fdce-4578-a830-dc1d11bee3d2)

![image](https://github.com/user-attachments/assets/dff150b3-c6a4-47b7-becb-88907a272336)

When the number of entities created in the World reaches about 500, the following error occurs. Why is this happening? Is there a limit on the total number of entities?

I am developing a game server, and I've created one World in a server and generated 101 map entities.

In each of the 101 maps, I created a World, and in each World, I generated about 500 entities.

**Comment by genaray** [comment]
Which Arch Version?

**Reply by demian-moon** [reply]
Hello Generay,
Arch is really amazing.
I'm always grateful to you.
The Arch version is 1.2.8 nuget package

And create more entities, including users, for each map, not just 500.

**Reply by genaray** [reply]
Have you tried the latest alpha version? 
Can you provide a minimal example that reproduces the error?

**Comment by demian-moon** [comment]
![image](https://github.com/user-attachments/assets/b8bb530c-c28b-4982-ab89-598ed02737c1)

Where can I find the latest alpha version?

Based on further testing, the Null Exception occurs regardless of the number of entities.

In what cases can an entity's World be null?
The entity has Id=27 and WorldId=14, but the World is null.

Though I’m not entirely sure, it seems that the entity whose Id is included in the RecycledEntity of recycledIds is causing the problem.

I apologize for not being able to reproduce the bug in a simple way.

**Comment by demian-moon** [comment]
![image](https://github.com/user-attachments/assets/69acb1a9-4b67-4d76-a507-ee75fb1a9229)

It's another error. The error occurs one level deeper here.

**Comment by demian-moon** [comment, marked as answer]
The bug occurred because I was careless with checking specific conditions, leading to the World.Destroy(Entity entity) function being executed more than once. I’m sorry for making you worry, especially when you must be busy
